http/client.go

```
@@ -14,46 +14,92 @@ import (
-}
+type ClientOption func(*http.Client) error
 
-// NewAcceptInvalidHostname new http.Client with ability to accept HTTPS certificates that don't
-// match the hostname of the server they are connecting to.
-func NewAcceptInvalidHostname(CABundleFile, CABundleDir string, httpTimeout time.Duration, hostname string) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, hostname)
-}
+// New creates a new http.Client with the Passed Client Options
+// that will have a custom certificate if it is loaded from the passed CA Bundle file and/or
+// directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
+func New(opts ...ClientOption) (*http.Client, error) {
+	t := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
 
-func _new(CABundleFile, CABundleDir string, httpTimeout time.Duration, acceptInvalidHostname string) (*http.Client, error) {
-	// go default http transport settings
-	t := &http.Transport{}
+	client := &http.Client{
+		Transport: t,
+	}
 
-	if CABundleFile != "" || CABundleDir != "" {
-		certs, err := getCertPool(CABundleFile, CABundleDir)
-		if err != nil {
+	for _, opt := range opts {
+		if err := opt(client); err != nil {
 			return nil, err
 		}
+	}
+
+	return client, nil
+}
+
+// WithCABundleFile adds the CABundleFile cert to the the client's certPool,
+// if a CABundleDir is passed it will be joined to the CABundleFile to try to get the file.
+// If the file is a full path, even it the CABundleDir is passed, it will be detected and the full
+// path will be used instead
+func WithCABundleFile(CABundleFile, CABundleDir string) ClientOption {
+	return func(c *http.Client) error {
+		if CABundleFile != "" {
+			certPool := getClientCertPool(c)
+			if err := addCert(filepath.Join(CABundleDir, CABundleFile), certPool); err != nil {
+				if os.IsNotExist(err) {
+					if err = addCert(CABundleFile, certPool); err != nil {
+						return err
+					}
+				}
+			}
+		}
+		return nil
```

We could return an error `if CABundleFile == ""` and reduce indentation.